lib/chef/knife/ssh.rb

```diff
@@ -298,6 +298,10 @@ def session_options(host, port, user = nil, gateway: false)
             opts[:keepalive] = true
             opts[:keepalive_interval] = ssh_config[:keepalive_interval]
           end
+          # maintain support for legacy key types / ciphers / key exchange algorithms.
+          # most importantly this adds back support for DSS host keys
+          # See https://github.com/net-ssh/net-ssh/pull/709
+          opts[:append_all_supported_algorithms] = true
```

Comment:
The comment on this method says it returns a hash, but this is going to return `true`.

```suggestion
          opts[:append_all_supported_algorithms] = true
          
          opts
```

Reply:
Nevermind, didn't notice this was in a `tap` block.